File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(const int x, const int y) {
    return (x%y==0)?y:gcd(y,x%y);
}

vector<int> solution(int n, int m) {
    int gc=gcd(n,m);
    vector<int> r={gc,n*m/gc};
    return r;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int gcd(int x,int y) {
        return x%y==0?y:gcd(y,x%y);
    }
    public int[] solution(int n, int m) {
        int gcdn=gcd(n,m), rlt[]={gcdn,n*m/gcdn};
        return rlt;
    }
}